disagmoe/models/experts.py

```diff
@@ -257,6 +258,179 @@ def forward(self, bs: int, hiddens: torch.Tensor, batch_sizes: torch.Tensor):
         return down_out
 
 
+class MoEExpertsDeepGemmFP8Graph(MoEExpertsDeepGemmFP8):
+    """DeepGEMM-based FP8 grouped experts using CUDAGraphs with bucketing."""
+
+    def __init__(
+        self,
+        hidden_size: int,
+        intermediate_size: int,
+        num_experts: int,
+        tp_size: int = 1,
+        max_batch_size: int = MAX_BATCH_SIZE,
+    ):
+        super().__init__(hidden_size, intermediate_size, num_experts, tp_size, max_batch_size)
+
+        self.graph_batch_sizes = self.get_graph_batch_sizes(max_batch_size)
+        self.graphs: Dict[int, torch.cuda.CUDAGraph] = {}
+        self.static_buffers: Dict[int, Dict[str, torch.Tensor]] = {}
+        
+        # Warmup and capture
+        self.capture_graphs()
+
+    def get_graph_batch_sizes(self, graph_max_batch_size: int) -> List[int]:
+        # Logic borrowed from CUDAGraphAttnExecutor
+        if graph_max_batch_size <= 1:
+            return [1]
+            
+        graph_bsz = [1]
+        # Adjust stages as needed. 
+        bsz_stage = [8, 128, 256, 512, 1024, 2048, 4096, 8192, 16384, 32768, 65536] 
+        bsz_inc = [0, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096]
```

Comment:
This bsz is too fine-grained.

just using 256, 512, 1024 might be good.

graph_bsz = [256, 512, 1024]